kube_dns/datadog_checks/kube_dns/kube_dns.py

```diff
@@ -2,77 +2,109 @@
-            # metrics names for kubernetes < 1.6.0
-            'skydns_skydns_dns_response_size_bytes': 'response_size.bytes',
-            'skydns_skydns_dns_request_duration_seconds': 'request_duration.seconds',
-            # Note: the count metrics were moved to specific functions below to be submitted as both gauges and monotonic_counts
+        # Set up metric_transformers
+        self.METRIC_TRANSFORMERS = {
+            'kubedns_kubedns_dns_request_count_total': self.kubedns_kubedns_dns_request_count_total,
+            'kubedns_kubedns_dns_error_count_total': self.kubedns_kubedns_dns_error_count_total,
+            'kubedns_kubedns_dns_cachemiss_count_total': self.kubedns_kubedns_dns_cachemiss_count_total,
+            'skydns_skydns_dns_request_count_total': self.skydns_skydns_dns_request_count_total,
+            'skydns_skydns_dns_error_count_total': self.skydns_skydns_dns_error_count_total,
+            'skydns_skydns_dns_cachemiss_count_total': self.skydns_skydns_dns_cachemiss_count_total
         }
 
+        # Create instances we can use in OpenMetricsBaseCheck
+        generic_instances = None
+        if instances is not None:
+            generic_instances = self.create_generic_instances(instances)
+
+        super(KubeDNSCheck, self).__init__(name, init_config, agentConfig, instances=generic_instances)
 
     def check(self, instance):
         endpoint = instance.get('prometheus_endpoint')
-        if endpoint is None:
-            raise CheckException("Unable to find prometheus_endpoint in config file.")
-
-        self.set_prometheus_timeout(instance)
+        scraper_config = self.config_map[endpoint]
+        self.process(scraper_config, metric_transformers=self.METRIC_TRANSFORMERS)
 
-        send_buckets = instance.get('send_histograms_buckets', True)
-        # By default we send the buckets.
-        if send_buckets is not None and str(send_buckets).lower() == 'false':
-            send_buckets = False
-        else:
-            send_buckets = True
+    def create_generic_instances(self, instances):
+        """
+        Transform each Kube DNS instance into a OpenMetricsBaseCheck instance
+        """
+        generic_instances = []
+        for instance in instances:
+            transformed_instance = self._create_kube_dns_instance(instance)
+            generic_instances.append(transformed_instance)
 
-        self.process(endpoint, send_histograms_buckets=send_buckets, instance=instance)
+        return generic_instances
 
-    def submit_as_gauge_and_monotonic_count(self, metric_suffix, message, **kwargs):
+    def _create_kube_dns_instance(self, instance):
+        """
+        Set up kube_dns instance so it can be used in OpenMetricsBaseCheck
+        """
+        # kube_dns uses 'prometheus_endpoint' and not 'prometheus_url', so we have to rename the key
+        instance['prometheus_url'] = instance.get('prometheus_endpoint', None)
```

Comment:
I don't know what the side effects of modifying the instance object could be (especially on agent5). Could we `deepcopy` it instead of modifying in place?